chore(base-driver): Remove the legacy JWP-related stuff

mykola-mokhnach opened the pull request:
## Proposed changes

Also deleted the hacky Python middleware. I assume it has been not relevant anymore for quite a while.

mykola-mokhnach committed:
chore(base-driver): Remove the legacy JWP-related stuff

mykola-mokhnach committed:
fix compilation

mykola-mokhnach committed:
Fix unit tests

mykola-mokhnach committed:
Update more tests

mykola-mokhnach commented:
Linking https://github.com/appium/appium/issues/20321

mykola-mokhnach committed:
Tune stacktrace

eglitise commented on packages/base-driver/lib/express/middleware.js:
```diff
@@ -5,6 +5,9 @@ export {handleIdempotency} from './idempotency';
 import {calcSignature} from '../helpers/session';
+import {getResponseForW3CError} from '../protocol/errors';
+
+const SESSION_ID_PATTERN = /\/session\/([^/]+)/;
```
This constant appears to be unused now

eglitise replied:
Nevermind, it is used on line 62.

eglitise reviewed (approved):
If I understand these changes correctly:
* Error messages will no longer include the `status` or `sessionId` fields, instead keeping them in `value` for a more standard W3C response
* Errors will also no longer include the `An unknown server-side error occurred while processing the command`/`The requested resource could not be found` parts and instead use the common `getResponseForW3CError` handler
* The `fixPythonContentType` hack has been removed and the header is instead explicitly set in `lib/protocol/protocol.js`

mykola-mokhnach commented:
> If I understand these changes correctly:
> 
> * Error messages will no longer include the `status` or `sessionId` fields, instead keeping them in `value` for a more standard W3C response

not all error responses had these extra fields, i've just cleaned then up from ones that still contained them

> * Errors will also no longer include the `An unknown server-side error occurred while processing the command`/`The requested resource could not be found` parts and instead use the common `getResponseForW3CError` handler

I've just removed the copy from middlewares as it was the same as the actual error message for corresponding errors types. I was going to create another PR afterwards to improve typing for these errors

> * The `fixPythonContentType` hack has been removed and the header is instead explicitly set in `lib/protocol/protocol.js`

headers tuning is not related to the middleware removal.

KazuCocoa reviewed (approved):
lg
